src/propulsion/calculate_RPM.cpp

```diff
@@ -0,0 +1,149 @@
+    log_(log),
+    motor_failure_(false),
+    slip_calculated_(false)
+{
+    state_      = data_.getStateMachineData();
+    motor_data_.module_status = data::ModuleStatus::kStart;
+    motor_data_.velocity_1 = 0;
+    motor_data_.velocity_2 = 0;
+    motor_data_.velocity_3 = 0;
+    motor_data_.velocity_4 = 0;
+    data_.setMotorData(motor_data_);
+    motor_velocity_ = {0, 0, 0, 0};
+    //communicator_ = new Communicator(log);
+}
+
+    void CalculateRPM::updateMotorFailure(){
+        log_.ERR("MOTOR", "Motor State: MOTOR FAILURE\n");
+        motor_data_ = data_.getMotorData();
+        motor_data_.module_status = data::ModuleStatus::kCriticalFailure;
+        data_.setMotorData(motor_data_);
+        motor_failure_ = true;
+    }
+
+    std::vector<std::vector<double>> CalculateRPM::transpose(std::vector<std::vector<double>> data){
+        std::vector<std::vector<double>> transpose(2, std::vector<double>(data.size(), 1));
+        for (uint16_t i = 0; i < data.size(); ++i) {
+            for (uint16_t j = 0; j < 2; ++j) {
+            transpose[j][i] = data[i][j];
+            }
+        }
+        return transpose; 
+    }
+
+    bool CalculateRPM::check_file(std::string filepath){
+
+        std::ifstream data;
+        data.open(filepath);
+
+        if (!data.is_open()) {
+            log_.ERR("MOTOR", "Could not open file: %s", filepath.c_str());
+            updateMotorFailure();
+            return false;
+        } else if (filepath == kAccelerationData) {
+            log_.INFO("MOTOR", "Calculating acceleration slip...");
+        } else {
+            log_.INFO("MOTOR", "Calculating deceleration slip...");
+        }
+
+        return true;
+
+    };
+
+    void CalculateRPM::RPM_calc(std::string filepath){
+
+        double slip, translational_velocity, angular_velocity, rpm;
+        std::ifstream data;
+        data.open(filepath);
+
+        std::string line;
+        while (std::getline(data, line)) {
+
+            std::string split_line;
+            std::vector<double> temp_vec;
+            std::stringstream ss(line);
+            while (std::getline(ss, split_line, '\t')) {
+                temp_vec.push_back(std::move(stod(split_line)));
+            }
+
+            // Calculate angular velocity from slip, translational velocity and halbach wheel radius
+            slip = temp_vec[0];
+            translational_velocity = temp_vec[1];
+            angular_velocity = (slip + translational_velocity) / kHalbachRadius;
+
+            // Calculate RPM given calculated angular velocity
+            rpm = (angular_velocity * 60) / (2*M_PI);
+
+            // Use temporary vector to hold translational velocity and rpm
+            temp_vec[0] = translational_velocity;
+            temp_vec[1] = rpm;
+
+            // Add data to appropriate container
+            if (filepath == kAccelerationData) {
+                acceleration_slip_.push_back(temp_vec);
+            } 
+            else {
```

Comment:
unneccessary if, since there is only acceleration data

Reply:
Removed condition.